src/main/java/org/tron/core/config/args/LocalWitnesses.java

```diff
@@ -33,17 +44,31 @@ public void setPrivateKeys(final List<String> privateKeys) {
     if (CollectionUtils.isEmpty(privateKeys)) {
       return;
     }
-    this.privateKeys = privateKeys;
     for (String privateKey : privateKeys) {
-      if (StringUtils.startsWithIgnoreCase(privateKey, "0X")) {
-        privateKey = privateKey.substring(2);
-      }
-
-      if (StringUtils.isNotBlank(privateKey) && privateKey.length() != 66) {
-        throw new IllegalArgumentException(
-            "Private key(" + privateKey + ") must be 66-bits hex string.");
-      }
+      validate(privateKey);
     }
+    this.privateKeys = privateKeys;
+  }
+
+  private void validate(String privateKey) {
+    if (StringUtils.startsWithIgnoreCase(privateKey, "0X")) {
+      privateKey = privateKey.substring(2);
+    }
+
+    if (StringUtils.isNotBlank(privateKey) && privateKey.length() != 66) {
+      throw new IllegalArgumentException(
+          "Private key(" + privateKey + ") must be 66-bits hex string.");
+    }
+  }
 
+  public void addPrivateKeys(String privateKey) {
+    validate(privateKey);
+    this.privateKeys.add(privateKey);
   }
+
+  //get the first one recently
+  public String getPrivateKey() {
```

Comment:
It's better to just call method with what it exactly does instead of adding additional comment -> a proposal `getFirstPrivateKey()` ? Are you sure it won't throw NPE?

Reply:
nice idea